### PR TITLE
Migrate OAuth HTTP code to new worker

### PR DESCRIPTION
**Pull request by mmastrac**
This migrates the auth extensions from httpx to the new Rust HTTP handler. We add a new, shared HttpClient to the tenant which anyone can make use of.

### CODE REVIEW

File: edb/server/http.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        if args:
+            data = args[0]
+            if isinstance(data, Mapping):
+                for key, value in data.items():
+                    self[key] = value
+            else:
+                for key, value in data:
+                    self[key] = value
+        for key, value in kwargs.items():
+            self[key] = value
```
Comment (msullivan) [inline edb/server/http.py:171]: Is it really worth doing this?

File: tests/test_http_ext_auth.py [hunk 2]
```diff
@@ -1417,8 +1417,8 @@ async def test_http_auth_ext_azure_authorize_01(self):
 
             discovery_request = (
                 "GET",
-                "https://login.microsoftonline.com/common/v2.0",
-                ".well-known/openid-configuration",
+                "https://login.microsoftonline.com",
+                "common/v2.0/.well-known/openid-configuration",
```
Comment (msullivan) [inline tests/test_http_ext_auth.py:1421]: Why do these need to change?

**Review by msullivan — APPROVED** [review]

File: tests/test_http_ext_auth.py (hunk 2 above)
Comment (mmastrac) [reply tests/test_http_ext_auth.py:1421]: This was relying on some very obscure parsing and encoding for base URLs. In the end I just cleaned up the URL handling so we didn't have to special case the base URL part of things.

File: edb/server/http.py (hunk 1 above)
Comment (mmastrac) [reply edb/server/http.py:171]: The getter is definitely necessary to make sure the older code works, but the setter is more for ensuring we don't hit any landmines with code expecting headers to work as it does in httpx. I can take another pass at this once I finish up the AI code and figure out exactly what level of compat is necessary.

**Pull request merged**